ADAL/src/workplacejoin/mac/ADWorkPlaceJoinUtil.m

```diff
@@ -51,58 +50,50 @@ + (ADRegistrationInformation *)getRegistrationInformation:(id<MSIDRequestContext
     NSString *certificateSubject = nil;
     NSData *certificateData = nil;
     NSString *certificateIssuer  = nil;
-    ADAuthenticationError *adError = nil;
-    
-    if (error)
-    {
-        *error = nil;
-    }
+    NSData *issuer = nil;
+    OSStatus status = noErr;
     
     MSID_LOG_VERBOSE(context, @"Attempting to get WPJ registration information");
+    identity = [self copyWPJIdentity:context issuer:&issuer certificateAuthorities:challenge.protectionSpace.distinguishedNames];
     
-    [self copyCertificate:&certificate identity:&identity issuer:&certificateIssuer context:context error:&adError];
-    if (adError)
+    // If there's no identity in the keychain, return nil. adError won't be set if the
+    // identity can't be found since this isn't considered an error condition.
+    if (!identity || CFGetTypeID(identity) != SecIdentityGetTypeID())
     {
-        if (error)
-        {
-            *error = adError;
-        }
-        MSID_LOG_ERROR(context, @"Failed to retrieve WPJ certificate. Error code: %ld", (long)adError.code);
+        MSID_LOG_VERBOSE(context, @"Failed to retrieve WPJ identity.");
         goto _error;
     }
     
-    // If there's no certificate in the keychain, return nil. adError won't be set if the
-    // cert can't be found since this isn't considered an error condition.
-    if (!certificate)
-    {
-        return nil;
-    }
+    // Get the wpj certificate
+    MSID_LOG_VERBOSE(context, @"Retrieving WPJ certificate reference.");
+    status = SecIdentityCopyCertificate(identity, &certificate);
+    CHECK_KEYCHAIN_STATUS(@"Failed to read WPJ certificate.");
```

Comment:
nit: i assume this returns nil and sets error if certificate not found?

Reply:
Yes the macro CHECK_KEYCHAIN_STATUS does that internally.